File: app/src/main/java/com/example/myapp/WeatherService/OpenWeatherRetrofit.java
package com.example.myapp.WeatherService;

import com.example.myapp.WeatherData.WeatherData;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface OpenWeatherRetrofit {
    String HTTP = "http";
    String HTTPS = "https";
    String BASE_URL = "://api.openweathermap.org/";
    String APP_ID = "bb18dcd129bad0dd351cdb2816a1aa9b";

    @GET("data/2.5/weather")
    Call<WeatherData> loadWeather(@Query("q") String location,
                                  @Query("appid") String appId,
                                  @Query("lang") String lang,
                                  @Query("units") String units);
}

File: app/src/main/java/com/chplalex/shaman/WeatherService/OpenWeatherRetrofit.kt
package com.chplalex.shaman.WeatherService

import com.chplalex.shaman.mvp.model.WeatherData
import io.reactivex.Observable
import io.reactivex.Single
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface OpenWeatherRetrofit {

    @GET("data/2.5/weather")
    fun loadWeather(
        @Query("q") location: String?,
        @Query("appid") appId: String?,
        @Query("lang") lang: String?,
        @Query("units") units: String?
    ): Call<WeatherData>

    @GET("data/2.5/weather")
    fun loadWeatherRX(
        @Query("q") location: String?,
        @Query("appid") appId: String?,
        @Query("lang") lang: String?,
        @Query("units") units: String?
    ): Single<WeatherData>

    companion object {

        const val HTTP = "http"
        const val HTTPS = "https"
        const val BASE_URL = "://api.openweathermap.org/"
        const val APP_ID = "bb18dcd129bad0dd351cdb2816a1aa9b"
    }
}